src/test/regress/input/temp_tablespaces.source

```diff
@@ -301,3 +301,156 @@ $$;
 drop table tmp_table_12871;
 drop tablespace space_12871;
 
+--
+-- temp_file_tablespaces:
+--
+SET optimizer = off;
+SET statement_mem = '2MB';
+
+CREATE TABLE tts_foo (a int, b int) DISTRIBUTED BY (a);
+CREATE TABLE tts_bar (c int, d int) DISTRIBUTED BY (d);
+CREATE TABLE tts_jazz (e int, f int) DISTRIBUTED BY (e);
+
+INSERT INTO tts_foo SELECT i, i FROM generate_series (1, 1000000) i;
+INSERT INTO tts_bar SELECT i, i FROM generate_series (6000, 120000) i;
+INSERT INTO tts_jazz SELECT i, i FROM generate_series (4000,150000) i;
+
+ANALYZE tts_foo;
+ANALYZE tts_bar;
+ANALYZE tts_jazz;
+
+-- If the temp_file_tablespaces is not empty, but temp_tablespaces is empty,
```

Comment:
how about to create function and call it for all cases?
```sql
create function report(
    global_temp_tablespaces text,
    global_temp_file_tablespaces text,
    session_temp_tablespaces text,
    session_temp_file_tablespaces text
) returns table ("temp table location" text, "temp files location" text) as $$
declare
begin
...
end;$$ language plpgsql;
select report(NULL, NULL, NULL, NULL);
select report('name1', NULL, NULL, NULL);
select report(NULL, 'name2', NULL, NULL);
select report('name1', 'name2', NULL, NULL);
... and so on ...
```

Reply:
What would be the differences between ` global_temp_...` and `session_temp_...`?

Reply:
global_temp is set by gpconfig while session_temp is set by set command

Reply:
Can you explain what else needs to be done and maybe provide an example?

Reply:
> maybe provide an example?

```sql
\! gpconfig -c temp_tablespaces -v name1
\! gpstop -u
```